apps/server/src/shared/infra/antivirus/antivirus.service.ts

```diff
@@ -14,9 +17,35 @@ interface AntivirusServiceOptions {
 export class AntivirusService {
 	constructor(
 		private readonly amqpConnection: AmqpConnection,
-		@Inject('ANTIVIRUS_SERVICE_OPTIONS') private readonly options: AntivirusServiceOptions
+		@Inject('ANTIVIRUS_SERVICE_OPTIONS') private readonly options: AntivirusServiceOptions,
+		private readonly clamConnection: NodeClam
 	) {}
 
+	public async checkStream(stream: Readable) {
+		const scanResult: ScanResultParams = {
+			virus_detected: undefined,
+			virus_signature: undefined,
+			error: undefined,
+		};
+		try {
+			const { isInfected, viruses } = await this.clamConnection.scanStream(stream);
+			if (isInfected === true) {
+				scanResult.virus_detected = true;
+				scanResult.virus_signature = viruses.join(',');
+			} else if (isInfected === null) {
+				scanResult.virus_detected = undefined;
+				scanResult.error = '';
+			} else {
+				scanResult.virus_detected = false;
+			}
+		} catch (error) {
+			scanResult.virus_detected = false;
```

Comment:
This catch is not correct. If someone write a js error in try block, or clamConnection go down, has wrong settings what ever, then a error must be throw. Full catching of errors is in any situation not what we want.

Reply:
Ich finde das legitim die Fehler nicht zu Thrown, da wir kein feedback an user geben wollen. Die Fehler soll auch ClamAV bei sich Thrown. Wir habe offt im Project das wir die Fehler anderes behandeln als bekommen.

https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/apps/server/src/modules/learnroom/service/board-copy.service.ts#L68-L74